draco/analysis/flagging.py

```diff
@@ -402,6 +406,205 @@ def process(self, timestream):
+            med = np.median(radiometer[:, ii][~start_flag])
+            mad = np.median(abs(radiometer[:, ii][~start_flag] - med))
+            rms = self.MADTORMS * mad
+            threshold1 = (
+                rms * self.start_threshold_sigma * self.max_m ** self.RMS_SCALING_DIFF
+            )
+
+            # SUMTHRESHOLD mask
+            stmask.append(
+                dracorfi.sumthreshold(
+                    radiometer[:, ii],
+                    self.max_m,
+                    start_flag=start_flag,
+                    threshold1=threshold1,
+                    correct_for_missing=True,
+                )
+            )
+
+        # Combine two polarization flags
+        madmask = np.logical_or(madmask[0], madmask[1])
+        stmask = np.logical_or(stmask[0], stmask[1])
+
+        if self.mask_type == "mad":
+            finalmask = madmask
+        elif self.mask_type == "sumthreshold":
+            finalmask = stmask
+        else:  # Combine ST and MAD masks
+            sidereal_day = ephemeris.SIDEREAL_S * 24 * 3600
+            timestamp = sensitivity.time
+            ntm = len(timestamp)
+            start_csd = ephemeris.unix_to_csd(timestamp[0])
+            start_ra = (start_csd - int(start_csd)) * (2.0 * np.pi)
+            ra_axis = (start_ra + (2.0 * np.pi) / ntm * np.arange(ntm)) % (2.0 * np.pi)
+            # Select Sun transit times
+            suntt = ephemeris.solar_transit(timestamp[0])  # Sun transit time
+            sunidx = np.argmin(abs(timestamp - suntt))
+            beam_window = (
+                2.0 * np.pi
+            ) / 90.0  # 4 deg in radians for the Sun. Each side.
+            # Mask the sun for the maximum aparent dec of ~24 deg
+            ra_window = beam_window / np.cos(np.deg2rad(24))
+            time_window = ra_window / (2 * np.pi) * sidereal_day
+            # Time spans where we apply the MAD filter.
+            madtimes = abs(timestamp - suntt) < time_window
+            # Select bright source transit times. Only CasA, CygA and TauA.
+            sources = [ephemeris.CasA, ephemeris.CygA, ephemeris.TauA]
+            # , ephemeris.VirA]
+            for src in sources:
+                src_ra = src.ra.radians
+                src_idx = np.argmin(abs(ra_axis - src_ra))
+                ra_window = beam_window / np.cos(src.dec.radians)
+                time_window = ra_window / (2 * np.pi) * sidereal_day
+                # Include bright point source transit in MAD times
+                madtimes += abs(timestamp - timestamp[src_idx]) < time_window
+            # Create final mask
+            finalmask = stmask
+            finalmask[:, madtimes] = madmask[:, madtimes]
+
+        # Apply scale invariant rank (SIR) operator, if asked for.
+        if self.sir:
+            finalmask = self.add_sir(finalmask, static_flag)
+
+        # Create container to hold mask
+        rfimask = containers.RFIMask(axes_from=sensitivity)
+        rfimask.mask[:] = finalmask
```

Comment:
Can properties be modified like this if you lack a `setter`?

Reply:
I am pretty sure they can. I think the way containers are defined you can directly assign to datasets in the root of the hdf5 file. Datasets inside groups are read only.
The slice operator `[:]` returns a view of the dataset array.

Reply:
Oh totally! This is what I meant: https://stackoverflow.com/questions/36029389/python-modifying-property-value

When you define a `@property`, I think it needs a corresponding `@setter` to be modifiable. But it is true that I am unsure how this plays out with NumPy views.

Reply:
I think you are right. I think the array view gets around that limitation because we don't assign to the property directly, but to a view of the array contained in the dataset?

Reply:
I think if you try rfimask.mask = some_array it would fail.

Reply:
Understood!